bundles/org.openhab.binding.kermi/src/main/java/org/openhab/binding/kermi/internal/api/KermiHttpUtil.java

```diff
@@ -0,0 +1,185 @@
+        this.hostname = hostname;
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    /**
+     * Issue a HTTP GET request and retry on failure
+     *
+     * @param url the url to execute
+     * @param timeout the socket timeout in milliseconds to wait for data
+     * @return the response body
+     * @throws KermiCommunicationException when the request execution failed or interrupted
+     */
+    @SuppressWarnings("null")
+    public synchronized String executeUrl(String httpMethod, String url, String content, String contentType)
+            throws KermiCommunicationException {
+        if (StringUtils.isBlank(hostname)) {
+            return "Not connected";
+        }
+
+        int attemptCount = 1;
+        try {
+            while (true) {
+                Throwable lastException = null;
+                String result = null;
+                try {
+                    InputStream inputStream = (content != null)
+                            ? new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8))
+                            : null;
+                    result = HttpUtil.executeUrl(httpMethod, url, httpHeaders, inputStream, contentType, 5000);
+                    logger.debug("[{} {}] {}", httpMethod, url, result);
+                } catch (IOException e) {
+                    // HttpUtil::executeUrl wraps InterruptedException into IOException.
+                    // Unwrap and rethrow it so that we don't retry on InterruptedException
+                    if (e.getCause() instanceof InterruptedException) {
+                        throw (InterruptedException) e.getCause();
+                    }
+
+                    if (e.getCause() instanceof ExecutionException) {
+                        ExecutionException iex = (ExecutionException) e.getCause();
+                        if (iex != null && iex.getCause() instanceof HttpResponseException) {
+                            HttpResponseException hre = (HttpResponseException) iex.getCause();
+                            if (401 == hre.getResponse().getStatus()) {
+                                logger.debug("Perform login");
+                                attemptCount = 0;
+                                performLogin();
+                            }
+                        }
+                    }
+                    lastException = e;
+                }
+
+                if (result != null) {
+                    if (attemptCount > 1) {
+                        logger.debug("Attempt #{} successful {}", attemptCount, url);
+                    }
+                    return result;
+                }
+
+                if (attemptCount >= 3) {
+                    logger.debug("Failed connecting to {} after {} attempts.", url, attemptCount, lastException);
+                }
+
+                logger.debug("HTTP error on attempt #{} {}", attemptCount, url);
+                Thread.sleep(500 * attemptCount);
```

Comment:
Are you sure you want to block the thread? why not use the scheduler? If this code is called on binding initilization, this is a no go. (i can check, but have to go now)